### PR TITLE
[WIP] ignore node ready condition as they're handled by taint manager

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**: This PR ignores the node Ready condition from being processed by taintByNodeCondition feature as this condition should be handled by the taint manager. Before this PR,  nodes were tainted as `NoSchedule` where they should be tainted as `NoExecute` (using `NoExecute` should be fine wrt pod safety)

**Which issue(s) this PR fixes**: Fixes #

**Special notes for your reviewer**:

/assign @wojtek-t 
cc @andrewsykim 

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by yastij** [comment]
failures are expected, we'll need to update taints related to `taintByNodeCondition`

File: pkg/controller/nodelifecycle/node_lifecycle_controller.go [hunk 1]
```diff
@@ -624,12 +624,15 @@ func (nc *Controller) doNoScheduleTaintingPass(nodeName string) error {
...
 	var taints []v1.Taint
 	for _, condition := range node.Status.Conditions {
-		if taintMap, found := nodeConditionToTaintKeyStatusMap[condition.Type]; found {
-			if taintKey, found := taintMap[condition.Status]; found {
-				taints = append(taints, v1.Taint{
-					Key:    taintKey,
-					Effect: v1.TaintEffectNoSchedule,
-				})
+		// ingnore nodeReady conditions as they're handled by the taint manager
+		if condition.Type != v1.NodeReady {
```
Comment (andrewsykim) [inline pkg/controller/nodelifecycle/node_lifecycle_controller.go:628]: This check should probably be conditional based on whether taint manager is running or not (see nc.runTaintManager).

**Comment by yastij** [comment]
@liggitt - added a commit to showcase how the test should be wired, as we were testing the wrong behaviour IMO. I'll also add another commit to test `TaintByNodeCondition` but without taint-manager enabled

**Comment by yastij** [comment]
ping @wojtek-t can you PTAL ?

File: pkg/controller/nodelifecycle/node_lifecycle_controller.go [hunk 2]
```diff
@@ -624,12 +624,15 @@ func (nc *Controller) doNoScheduleTaintingPass(nodeName string) error {
...
 	var taints []v1.Taint
 	for _, condition := range node.Status.Conditions {
-		if taintMap, found := nodeConditionToTaintKeyStatusMap[condition.Type]; found {
-			if taintKey, found := taintMap[condition.Status]; found {
-				taints = append(taints, v1.Taint{
-					Key:    taintKey,
-					Effect: v1.TaintEffectNoSchedule,
-				})
+		// ingnore nodeReady conditions as they're handled by the taint manager
+		if nc.runTaintManager && condition.Type != v1.NodeReady {
```
Comment (michaelgugino) [inline pkg/controller/nodelifecycle/node_lifecycle_controller.go:628]: This is confusing.  Is this code not part of the taintManager itself?  If this code is part of taintManager, then saying 'nodeReady conditions handled by taint manager' doesn't make sense.

If this code is not the taint manager, then what should happen if runTaintManager is false?  Who will be setting these taints?

Comment (mrajashree) [reply pkg/controller/nodelifecycle/node_lifecycle_controller.go:628]: I have the same question ^. Also the bug says that Unreachable taint with NoExecute effect is not set for NodeReady=Unknown. With this change even the Unreachable=NoSchedule taint won't get set?

**Pull request closed without merging**